### PR TITLE
EvaluateRubricJob: fetch user source code, and write fake data to LearningGoalAiEvaluation

### PR DESCRIPTION
**Pull request by davidsbailey**
Depends on https://github.com/code-dot-org/code-dot-org/pull/54086 . Continues https://codedotorg.atlassian.net/browse/AITT-132. 

Done in this PR:
- get channel id containing the user's source code on the given script level
- fetch user's source code on S3
- generate fake openai response
- write LearningGoalAiEvaluations with fake values for understandings

## Screenshots

https://github.com/code-dot-org/code-dot-org/assets/8001765/0fd08ec8-9f6b-4aa3-9e23-071531bf9180

## Testing story

* new unit tests cover main success case plus failure paths
* manually verified end-to-end as shown in screenshots
* mostly-passing drone run with just one unrelated / flaky failure: https://drone.cdn-code.org/code-dot-org/code-dot-org/34136

## Deployment strategy

Seems safe to deploy now that https://github.com/code-dot-org/code-dot-org/pull/53976 has been merged (one delayed job worker). Because the long-running call to openai has been stubbed out, one worker should be sufficient to cover any incidental testing of this feature on production. additionally, if anything does go wrong with the jobs, nothing will break besides this in-development feature.

https://github.com/code-dot-org/code-dot-org/pull/53976 is important because without it, we'd be using the async adapter, which could tie up the worker threads on production frontends.

### CODE REVIEW

File: dashboard/app/jobs/evaluate_rubric_job.rb [hunk 1]
```diff
@@ -30,7 +30,15 @@ def perform(user_id:, script_level_id:)
...
     raise "lesson_s3_name not found for script_level_id: #{script_level.id}" if lesson_s3_name.blank?
 
-    puts "TODO: Implement the rest of this job"
+    channel_id = get_channel_id(user, script_level)
+    code, project_version = read_user_code(channel_id)
+    puts "code: #{code.inspect}"
+
+    rubric = Rubric.find_by!(lesson_id: script_level.lesson.id, level_id: script_level.level.id)
+
+    ai_evaluations = get_fake_openai_evaluations(rubric, understanding_s: 'Extensive Evidence')
```
Comment (Nokondi) [inline dashboard/app/jobs/evaluate_rubric_job.rb:43]: lol is this why I only got "Extensive Evidence" during the bug bash last week?

**Review by Nokondi — APPROVED** [review]

File: dashboard/app/jobs/evaluate_rubric_job.rb (hunk 1 above)
Comment (davidsbailey) [reply dashboard/app/jobs/evaluate_rubric_job.rb:43]: haha, that WOULD explain it ! but no, this code was never on the adhoc. the adhoc was using pycall to make real calls to openai. this is just a tactic to break up the work and get all the non-python stuff merged while we figure out how to move from pycall to python service.

File: dashboard/app/jobs/evaluate_rubric_job.rb [hunk 2]
```diff
@@ -26,13 +26,18 @@ def perform(user_id:, script_level_id:)
...
-    end
 
     raise "lesson_s3_name not found for script_level_id: #{script_level.id}" if lesson_s3_name.blank?
 
-    puts "TODO: Implement the rest of this job"
+    channel_id = get_channel_id(user, script_level)
+    code, project_version = read_user_code(channel_id)
+    puts "code: #{code.inspect}"
+
+    rubric = Rubric.find_by!(lesson_id: script_level.lesson.id, level_id: script_level.level.id)
```
Comment (bethanyaconnor) [inline dashboard/app/jobs/evaluate_rubric_job.rb:41]: should this return if there isn't a rubric?

Comment (bethanyaconnor) [reply dashboard/app/jobs/evaluate_rubric_job.rb:41]: oh I see, this will intentionally throw an exception

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**